riscv-configuration-structure-draft.adoc

```diff
@@ -268,9 +268,37 @@ big immediate benefit of a format that supports comments.
 [[sec:AccessMethod]]
 == Access Method
 
-The binary configuration structure is memory-mapped in system memory.
-There is a CSR which contains the physical address where the structure
-starts.
+The binary configuration structure is memory-mapped in system memory.  There is
+a CSR which contains the physical address where the structure starts.
+
+TODO: Get a CSR number assigned.
+
+When software running on a hart wants to read the configuration structure, it
+reads this CSR, and then decodes the binary structure at that physical address.
+The structure will specify which parts apply to which harts (identified by hart
+ID), and the software can ignore any information that does not apply to the hart
+it's running on.
+
+This provides hardware implementers several options for embedding the binary
+structure:
+1. The structure can describe all harts, and be accessible over the memory bus.
+All harts have the same memory map and the same CSR contents.
+2. There might be several structures in the system, and different harts are
+pointed to different structures by having different memory maps.
+3. There might be several structures in the system, and different harts are
+pointed to different structures by having different pointers in their CSR.
```

Comment:
What is the difference between these two? one is "having different memory maps" and another is "having different pointers in their CSR". The pointer of the memory map is retrieved from CSR, thus I think #2 and #3 use cases are the same. no?

Reply:
Another method:
There might be several structures in the system, and different harts are pointed to different structures by having the same pointer in their CSR, however, the memory map pointed by CSR is only visible to the local hart.

Reply:
I think what you just wrote is what I intended by option 2. Perhaps we don't mean the same thing by "memory map." I'll try to rephrase that.

Reply:
The method I pointed out is the "different harts are pointed to different structures by having the **same**  memory map address". However, the memory map is only visible to the local hart.
These three methods are actually implementation-specific, we just list some design examples.

Reply:
We're definitely talking about the same thing, we just need to figure out language that makes that clear to everybody. How about:

"There might be several structures in the system. Each hart has the same address in mconfigptr. The memory system provides a different configuration structure at that address depending on which hart is performing the access."

I don't really want to use the word "memory map" because to me it means something like the picture in http://www.sharetechnote.com/html/EmbeddedSystem_MemoryMap.html which is not really relevant here.

Reply:
yes, this sounds to me better. thanks.